kong/runloop/plugins_iterator.lua

```diff
@@ -157,28 +157,28 @@ local function load_configuration_through_combos(ctx, combos, plugin)
   local  service_id = service  and  service.id or nil
   local consumer_id = consumer and consumer.id or nil
 
-  if route_id and service_id and consumer_id and combos[COMBO_RSC] then
+  if route_id and service_id and consumer_id and combos[COMBO_RSC] and combos.both[route_id] == service_id then
```

Comment:
Could we limit the line length appropriately in those conditions now, please?

Reply:
line length limited to max 80 chars.